pkg/storaged/btrfs/subvolume.jsx

```diff
@@ -0,0 +1,106 @@
+import client from "../client.js";
+
+const _ = cockpit.gettext;
+
+function subvolume_unmount(volume, subvol, mount_point) {
+    const block = client.blocks[volume.path];
+    // HACK: generalise this? Instead of a global findmnt state, put the active mount point in subvol.
+    subvol.mount_point = mount_point;
+    mounting_dialog(client, block, "unmount", null, subvol);
+}
+
+function subvolume_mount(volume, subvol) {
+    const block = client.blocks[volume.path];
+    mounting_dialog(client, block, "mount", null, subvol);
+}
+
+export function make_btrfs_subvolume_page(parent, volume, subvol) {
+    const actions = [];
+
+    const use = btrfs_usage(client, volume);
+    const backing_block = client.blocks[volume.path];
+    const content_block = client.blocks_fsys[volume.path];
+    const fstab_config = get_fstab_config_with_client(client, backing_block, false, subvol);
+    const [, mount_point] = fstab_config;
+    const mismount_warning = check_mismounted_fsys(backing_block, content_block, fstab_config, subvol);
+    const mounted = is_mounted(client, backing_block, subvol);
+    const mp_text = mount_point_text(mount_point, mounted);
+
+    if (mounted) {
+        actions.push({
+            title: _("Unmount"),
+            action: () => subvolume_unmount(volume, subvol, mount_point),
+        });
+    } else {
+        actions.push({
+            title: _("Mount"),
+            action: () => subvolume_mount(volume, subvol),
+        });
+    }
+
+    const card = new_card({
+        title: _("btrfs subvolume"),
+        next: null,
+        // HACK: page_location does not work well with a path that contains a Number, likely due to JSON.stringify usage?
+        page_location: ["btrfs", volume.data.uuid, String(subvol.id)],
+        page_name: subvol.pathname,
+        page_size: is_mounted && <StorageUsageBar stats={use} short />,
+        location: mp_text,
+        component: BtrfsSubvolumeCard,
+        has_warning: !!mismount_warning,
+        props: { volume, subvol, mount_point, mismount_warning, backing_block, content_block, fstab_config },
+        actions,
+    });
+    new_page(parent, card);
+}
+
+const BtrfsSubvolumeCard = ({ card, volume, subvol, mismount_warning, backing_block, content_block, fstab_config }) => {
+    return (
+        <StorageCard card={card} alert={mismount_warning &&
+        <MismountAlert warning={mismount_warning}
+                                    fstab_config={fstab_config}
+                                    backing_block={backing_block} content_block={backing_block} subvol={subvol} />}>
+            <CardBody>
+                <DescriptionList className="pf-m-horizontal-on-sm">
+                    <StorageDescription title={_("Name")} value={subvol.pathname} />
+                    <StorageDescription title={_("ID")} value={subvol.id} />
+                    <StorageDescription title={_("Mount point")}>
+                        <MountPoint fstab_config={fstab_config}
```

Comment:
You need to pass the subvol to `MountPoint` and have it pass it on to `is_mounted`.

Reply:
Nicely spotted!

Reply:
Also `forced_options`.